roles/reset/tasks/main.yml

```diff
@@ -237,6 +237,18 @@
     - enable_nodelocaldns|default(false)|bool
     - nodelocaldns_device.stat.exists
 
+- name: reset | gather /etc/hosts backup files
+  shell: set -o pipefail && find /etc -maxdepth 1 -type f -name 'hosts.*@*~'
```

Comment:
you don't need `set -o pipefail` if there is no pipe
you could use `find:`, or better, use `file:` and `with_fileglob`

Reply:
`with_fileglob` only work on ansible localhost
if use `find` will introduce a task to delete these files, like this:
```yml
- name: reset | gather /etc/hosts backup files
  find:
    path: /etc
    use_regex: yes
    patterns: "hosts.*@*~"
  register: hosts_backup_files
  tags:
    - files
    - dns

- name: reset | delete /etc/hosts backup files
  file:
    path: "{{ item.path }}"
    state: absent
  with_items:
    - "{{ hosts_backup_files.files }}"
  ignore_errors: yes
  tags:
    - files
    - dns
```

use shell to gather these files then delete it in `reset | delete some files and directories` task, this way can be done simply

I'm a bit confused, which method should be selected as appropriate 🤔️

Reply:
I completely forgot that with_fileglob is local only ...
You can use `"{{ hosts_backup_files.files | map(attribute='path') }}"` to get the same result as your shell
but I'm not sure your current code works because you are adding a list, not an item, into the list of file to delete

Reply:
It doesn't seem to work :-(

```bash
ok: [node-1] => (item=<generator object do_map at 0x7fdad2091308>)
ok: [node-2] => (item=<generator object do_map at 0x7fdad2091308>)
ok: [node-3] => (item=<generator object do_map at 0x7fdad2091308>)
ok: [node-4] => (item=<generator object do_map at 0x7fdad2091308>)
```